irr/src/COpenGLCoreTexture.h

```diff
@@ -422,6 +422,43 @@ class COpenGLCoreTexture : public ITexture
 		Driver->getCacheHandler()->getTextureCache().set(0, prevTexture);
 	}
 
+	void drawToSubImage(int x, int y, int width, int height, ITexture *texture) override
```

Comment:
wrong abstraction for this function

calling `lock()` on an `ITexture` will download the pixel data back from the GPU, which is a pointless waste of time.
this should either
* do the copy on the gpu
* take an `IImage` instead

Reply:
I've fixed it just getting the first cached image from the new `ITexture::getImagesCache()` and drawing its data